packages/donations-app/app/components/checkout/checkoutController.js

```diff
@@ -1,69 +1,34 @@
 angular.module('aliceApp')
-  .controller('CheckoutController', ['$rootScope', '$uibModal', '$http', '$stateParams', '$state', '$sce', '$timeout', 'AuthService', 'ProjectService', 'NotificationService', 'API', '$scope', 'MANGO', function ($rootScope, $uibModal, $http, $stateParams, $state, $sce, $timeout, AuthService, ProjectService, NotificationService, API, $scope, MANGO) {
+  .controller('CheckoutController', ['$rootScope', '$uibModal', '$http', '$stateParams', '$state', '$sce', '$timeout', 'AuthService', 'ProjectService', 'NotificationService', 'API', '$scope', 'MANGO', 'CheckoutService', function ($rootScope, $uibModal, $http, $stateParams, $state, $sce, $timeout, AuthService, ProjectService, NotificationService, API, $scope, MANGO, CheckoutService) {
 
-    // Set MANGOPAY API base URL and Client ID
-    mangoPay.cardRegistration.baseURL = MANGO.url;
-    mangoPay.cardRegistration.clientId = MANGO.clientId;
-
-    var amounts = {
+    const amounts = {
       amount10: 1000,
-      amount25: 2500,
-      amount50: 5000,
-      amount100: 10000,
+      amount20: 2000,
+      amount80: 8000,
+      amount150: 15000,
       amountOther: 0
     };
 
-
     var vm = this;
     vm.validatedGuest = false;
     vm.guest = {};
     vm.auth = AuthService;
-    vm.mode = 'CARD'; // enum: ["CARD", "3DS", "BANK_TRANSFER", "BANK_TRANSFER_REQUESTED"]
-
-    vm.card = {};
-
-    if (MANGO.url == 'https://api.sandbox.mangopay.com') {
-      vm.card = {
-        number: "3569990000000132",
-        expiryDate: "11/20",
-        cvc: "123",
-        name: AuthService.getLoggedFullName()
-      };
-    }
-
-    ProjectService.getProjectDetails($stateParams.projectCode).then(function (project) {
-      vm.project = project.data;
-    });
 
-    if (AuthService.getLoggedUser()) {
-      vm.card.name = AuthService.getLoggedUser().firstName + " " + AuthService.getLoggedUser().lastName;
-      vm.user = AuthService.getLoggedUser();
-    }
+    //TODO: Consider binding that to CheckoutService.donation.type, after we've got the bank mode
+    vm.mode = 'CARD';  // enum: ["CARD", "3DS", "BANK_TRANSFER", "BANK_TRANSFER_REQUESTED"]
 
-    $rootScope.$on('user:login', function (event, data) {
-      vm.card.name = AuthService.getLoggedFullName();
-      vm.user = AuthService.getLoggedUser();
-      if (!vm.user._id && vm.user.address1) {
-        AuthService.getLoggedUser().address1 = vm.user.address1;
-        AuthService.getLoggedUser().address2 = vm.user.address2;
-        AuthService.getLoggedUser().city = vm.user.city;
-        AuthService.getLoggedUser().postCode = vm.user.postCode;
-      }
-    });
-
-    var rescale = function () {
-      vm.iframeScale = Math.min(1, window.innerWidth / 500);
+    vm.setMode = function (mode) {
+      vm.mode = mode;
     };
 
-    rescale();
-
-    $(window).resize(function () {
-      $scope.$apply(rescale);
-    });
+    vm.card = CheckoutService.card;
```

Comment:
We can use lodash with "cloneDeep" and "pick" functions here

Reply:
But I don't want to clone but to reference it as the same object